node_modules/gh-users/tests/test-search.js

```diff
@@ -0,0 +1,214 @@
+var moment = require('moment');
+
+var EventsTestsUtil = require('gh-events/tests/util');
+var OrgUnitTestUtil = require('gh-orgunit/tests/util');
+var SeriesTestUtil = require('gh-series/tests/util');
+var TestsUtil = require('gh-tests');
+
+var UsersTestsUtil = require('./util');
+
+
+describe('Users - Search', function() {
+
+    /**
+     * Create a couple of users with distinct names and email address
+     *
+     * @param  {Function}       callback            Standard callback function
+     * @param  {Object}         callback.jack       User object as generated by `TestsUtil.generateTestUsers`
+     * @param  {Object}         callback.william    User object as generated by `TestsUtil.generateTestUsers`
+     * @param  {Object}         callback.shirley    User object as generated by `TestsUtil.generateTestUsers`
+     */
+    var _setup = function(callback) {
+        TestsUtil.generateTestUsers(global.tests.apps.cam2013, 3, false, function(jack, william, shirley) {
+
+            // Update the display names and email addresses
+            UsersTestsUtil.assertUpdateUser(jack.client, jack.profile.id, {'displayName': 'Jack Daniels', 'email': TestsUtil.generateTestEmailAddress('jd123')}, function() {
+                UsersTestsUtil.assertUpdateUser(william.client, william.profile.id, {'displayName': 'William Lawson', 'email': TestsUtil.generateTestEmailAddress('wl4567')}, function() {
+                    UsersTestsUtil.assertUpdateUser(shirley.client, shirley.profile.id, {'displayName': 'Shirley Temple', 'email': TestsUtil.generateTestEmailAddress('st789')}, function() {
+
+                        return callback(jack, william, shirley);
+                    });
+                });
+            });
+        });
+    };
+
+    /**
+     * Test that verifies that users can be queried on their display name
+     */
+    it('verify users can be queried on their display name', function(callback) {
+        _setup(function(jack, william, shirley) {
+
+            // Search for 'jac'
+            UsersTestsUtil.getUsers(global.tests.admins.cam2013.client, null, 'jac', 50, null, function(users) {
+                assert.ok(users.results.length > 0);
+
+                // Get the index of our three users in the results
+                var jackIndex = _.findIndex(users.results, {'id': jack.profile.id});
+                var williamIndex = _.findIndex(users.results, {'id': william.profile.id});
+                var shirleyIndex = _.findIndex(users.results, {'id': shirley.profile.id});
+
+                // Because the endpoint only returns x number of users, The "William Lawson" and "Shirley Temple"
+                // user might not get returned. Default the index value to a really high number, so we can more easily
+                // reason about them
+                jackIndex = (jackIndex === -1) ? Number.MAX_VALUE : jackIndex;
+                williamIndex = (williamIndex === -1) ? Number.MAX_VALUE : williamIndex;
```

Comment:
Although unlikely, it can happen. The email field is randomised and could potentially have "jac" (or something similar to it) in it.